airbyte-integrations/bases/source-test-lib/src/main/java/io/airbyte/integrations/base/TestSource.java

```diff
@@ -0,0 +1,265 @@
+
+  /**
+   * Name of the docker image that the tests will run against.
+   *
+   * @return docker image name
+   */
+  protected abstract String getImageName();
+
+  /**
+   * Configuration specific to the integration. Will be passed to integration where appropriate in
+   * each test. Should be valid.
+   *
+   * @return integration-specific configuration
+   */
+  protected abstract JsonNode getConfig() throws Exception;
+
+  /**
+   * Catalog to be used when attempting read operations.
+   *
+   * @return the catalog
+   * @throws Exception - do what must be done.
+   */
+  protected abstract AirbyteCatalog getCatalog() throws Exception;
+
+  /**
+   * Function that performs any setup of external resources required for the test. e.g. instantiate a
+   * postgres database. This function will be called before EACH test.
+   *
+   * @param testEnv - information about the test environment.
+   * @throws Exception - can throw any exception, test framework will handle.
+   */
+  protected abstract void setup(TestDestinationEnv testEnv) throws Exception;
+
+  /**
+   * Function that performs any clean up of external resources required for the test. e.g. delete a
+   * postgres database. This function will be called after EACH test. It MUST remove all data in the
+   * destination so that there is no contamination across tests.
+   *
+   * @param testEnv - information about the test environment.
+   * @throws Exception - can throw any exception, test framework will handle.
+   */
+  protected abstract void tearDown(TestDestinationEnv testEnv) throws Exception;
+
+  @Before
+  public void setUpInternal() throws Exception {
+    Path testDir = Path.of("/tmp/airbyte_tests/");
+    Files.createDirectories(testDir);
+    final Path workspaceRoot = Files.createTempDirectory(testDir, "test");
+    jobRoot = Files.createDirectories(Path.of(workspaceRoot.toString(), "job"));
+    localRoot = Files.createTempDirectory(testDir, "output");
+    testEnv = new TestDestinationEnv(localRoot);
+
+    setup(testEnv);
+
+    pbf = new DockerProcessBuilderFactory(
+        workspaceRoot,
+        workspaceRoot.toString(),
+        localRoot.toString(),
+        "host");
+  }
+
+  @After
+  public void tearDownInternal() throws Exception {
+    tearDown(testEnv);
+  }
+
+  /**
+   * Verify that when the integrations returns a valid spec.
+   */
+  @Test
+  public void testGetSpec() {
+    final OutputAndStatus<StandardGetSpecOutput> output = runSpec();
+    assertTrue(output.getOutput().isPresent());
+  }
+
+  /**
+   * Verify that when given valid credentials, that check connection returns a success response.
+   * Assume that the {@link TestSource#getConfig()} is valid.
+   */
+  @Test
+  public void testCheckConnection() throws Exception {
+    final OutputAndStatus<StandardCheckConnectionOutput> output = runCheck();
+    assertTrue(output.getOutput().isPresent());
+    assertEquals(Status.SUCCEEDED, output.getOutput().get().getStatus());
+  }
+
+  // /**
```

Comment:
currently we don't pass in enough information to ru this test, but i'd like to add it soon.